lib/galaxy/config/__init__.py

```diff
@@ -283,7 +283,8 @@ def strip_deprecated_dir(key, value):
         for key, value in kwargs.items():
             if key in self.appschema:
                 value = convert_datatype(key, value)
-                value = strip_deprecated_dir(key, value)
+                if value:
```

Comment:
```suggestion
                if value is None:
```
Based on the description that seems more correct to me ?

Reply:
It should be `if value` (or `if value is not None`) . Sorry - my description does sound confusing: I meant that it *may* be None, in which case the code breaks, so this is a fix (because the code will try to split a None value). 
That said, while looking into this I found a case that would break things: value may be not a string (if set by mistake). I'll test for more edge cases and will submit a fix later today.

Reply:
I checked. In this case it cannot be a wrong datatype: so no additional bug. 
May I mark as resolved?